### PR TITLE
Phase Banner

### PR DESCRIPTION
**Pull request by DanielClarkeEducation**
# Description

- Added new phase banner component stuck to just below the header

## Ticket number (if applicable)

https://dfedigital.atlassian.net/browse/EYQB-231

# How Has This Been Tested?

- manual testing locally (see screenshot)
- new unit and E2E tests passing.

# Screenshots

image

# Checklist:

- [x] My code follows the standards used within this project
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests (Unit, E2E) that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Dfe.EarlyYearsQualification.Content/Entities/PhaseBanner.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+using Contentful.Core.Models;
+
+namespace Dfe.EarlyYearsQualification.Content.Entities;
+
+public class PhaseBanner
+{
+  public string PhaseName { get; init; } = string.Empty;
+
+  public Document? Content { get; set; }
```
Comment (RobertGHippo) [inline src/Dfe.EarlyYearsQualification.Content/Entities/PhaseBanner.cs:9]: suggestion: can be `init` (don't want code to be able to change this value)

File: src/Dfe.EarlyYearsQualification.Content/Entities/PhaseBanner.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+  public string ContentHtml {get; set; } = string.Empty;
```
Comment (RobertGHippo) [inline src/Dfe.EarlyYearsQualification.Content/Entities/PhaseBanner.cs:11]: question: should this be `init`?

We don't want code to be able to change this? The property would then have to be either nullable `string?`, or settable by constructor parameter.

File: src/Dfe.EarlyYearsQualification.Content/Services/ContentfulContentService.cs [hunk 3]
```diff
@@ -140,6 +141,19 @@ public ContentfulContentService(IContentfulClient contentfulClient, ILogger<Cont
...
+    public async Task<PhaseBanner?> GetPhaseBannerContent()
+    {
+      var phaseBannerEntities = await GetEntriesByType<PhaseBanner>();
+      if (phaseBannerEntities is null || !phaseBannerEntities.Any())
+      {
+          _logger.LogWarning("No phase banner entry returned");
+          return default;
+      }
+      var phaseBannerContent = phaseBannerEntities.First();
+      phaseBannerContent.ContentHtml = await GetPhaseBannerRenderer().ToHtml(phaseBannerContent.Content);
```
Comment (RobertGHippo) [inline src/Dfe.EarlyYearsQualification.Content/Services/ContentfulContentService.cs:153]: suggestion: HTML-rendering logic should not be the responsibility of this service?

Single responsibility principle: I'd like to suggest that this service should be responsible only for getting the content from Contentful? I think the rendering would be better done later and elsewhere. With this current pattern, it's possible for the C# code to change the value of the `ContentHtml` on the object after it's been returned from this service. Wouldn't it be better for the `PhaseBanner` object not to have a `ContentHtml` property at all, and for the renderer to calculate it at just the point it is needed?

I realise this is a discussion point, and if we all agree on this we'd want to refactor the way this is done elsewhere too. I'd be happy to do that myself in another branch/PR... :)

File: tests/Dfe.EarlyYearsQualification.UnitTests/Mocks/MockContentfulServiceTests.cs
Comment (RobertGHippo) [inline tests/Dfe.EarlyYearsQualification.UnitTests/Mocks/MockContentfulServiceTests.cs:1]: suggestion: move this class into namespace `Dfe.EarlyYearsQualification.UnitTests.Mocks`

…to match file location

File: tests/Dfe.EarlyYearsQualification.UnitTests/Mocks/MockContentfulServiceTests.cs [hunk 4]
```diff
@@ -123,4 +123,15 @@ public async Task GetStartPage_ReturnsExpectedDetails()
...
     }
+
+    [TestMethod]
+    public async Task GetPhaseBannerContent_ReturnsExpectedDetails()
+    {
+        var result = await _contentfulService.GetPhaseBannerContent();
+        result.Should().NotBeNull();
+        result.Should().BeAssignableTo<PhaseBanner>();
+        result!.ContentHtml.Should().NotBeNullOrEmpty();
+        result!.PhaseName.Should().NotBeNullOrEmpty();
```
Comment (RobertGHippo) [inline tests/Dfe.EarlyYearsQualification.UnitTests/Mocks/MockContentfulServiceTests.cs:134]: suggestion: no need to assert it's not null (`!` is redundant) except the first time

[the same elsewhere in this class]

File: src/Dfe.EarlyYearsQualification.Content/Entities/PhaseBanner.cs (hunk 1 above)
Comment (sam-c-dfe) [reply src/Dfe.EarlyYearsQualification.Content/Entities/PhaseBanner.cs]: Fixed now

File: tests/Dfe.EarlyYearsQualification.UnitTests/Mocks/MockContentfulServiceTests.cs
Comment (sam-c-dfe) [reply tests/Dfe.EarlyYearsQualification.UnitTests/Mocks/MockContentfulServiceTests.cs:1]: Updated now :-)

File: tests/Dfe.EarlyYearsQualification.UnitTests/Mocks/MockContentfulServiceTests.cs (hunk 4 above)
Comment (sam-c-dfe) [reply tests/Dfe.EarlyYearsQualification.UnitTests/Mocks/MockContentfulServiceTests.cs]: Resolved all these in the class now

File: src/Dfe.EarlyYearsQualification.Content/Services/ContentfulContentService.cs (hunk 3 above)
Comment (sam-c-dfe) [reply src/Dfe.EarlyYearsQualification.Content/Services/ContentfulContentService.cs:153]: We have a chat about this later

**Review by RobertGHippo — APPROVED** [review]

File: src/Dfe.EarlyYearsQualification.Content/Services/ContentfulContentService.cs (hunk 3 above)
Comment (RobertGHippo) [reply src/Dfe.EarlyYearsQualification.Content/Services/ContentfulContentService.cs:153]: Yes, this is a topic for its own branch & PR

**Pull request merged**